### ISSUE TITLE
Optional protocol extensions break build for type Any?

### ISSUE THREAD

**Issue by wassil** [MakeAWishFoundation/SwiftyMocky#235]
After importing SwiftyMocky, code `let something: Any? = 123` doesn't compile, the error is "Protocol type 'Any' cannot conform to 'ExpressibleByIntegerLiteral' because only concrete types can conform to protocols"

`let something: Any = 123` or `let something: Int? = 123` compiles without issues

I've noticed SwiftyMocky creates protocol extensions on Optional types, seems like a root cause of this.

**Comment by sugarbaron** [comment]
I have the same issue. 
In my case, even when Any is not optional, and i'm trying to initialize it, i'm getting this compilation error.

for example:
let parameters: [String : Any] = ["name" : "value"] // compiler error for "value"

**Comment by amichnia** [comment]
Ok, I'll look into it. Could you, at least for now, try to cast it manually?
I think this would workaround it for now:
```swift
let parameters: [String : Any] = ["name" : "value" as Any] 
```

**Comment by g-amichnia** [comment]
Ok, I have mixed feelings about it. Seems more like a Swift issue for me. I started a thread on swift.org, hope I can get some clarification there.

Anyway, I will restrict these literals by a compile time flag for now. It would be automatically on for tests, and off for prototyping (where it actually can make some mess). 

In general - I still believe it is ok in test targets, using workarounds like:

```swift
// 1. Cast
let something: Any? = 123 as Any
// 2. Use init
let something: Any? = .init(123)
```
